### DISCUSSION TITLE
sokol API changes

### DISCUSSION THREAD

**Discussion by jorgeluismireles** [vlang/v#23082] [Questions and Answers] [answered]
I have code that worked with [sokol_gp](https://github.com/edubart/sokol_gp) but last commits was updated with the latest sokol. The API has changed, for instance the file's `sokol_gfx.h` struct `sg_shader_desc` now is
https://github.com/floooh/sokol/blob/master/sokol_gfx.h#L3175 :
```c
typedef struct sg_shader_desc {
    uint32_t _start_canary;
    sg_shader_function vertex_func;
    sg_shader_function fragment_func;
    sg_shader_vertex_attr attrs[SG_MAX_VERTEX_ATTRIBUTES];
    sg_shader_uniform_block uniform_blocks[SG_MAX_UNIFORMBLOCK_BINDSLOTS];
    sg_shader_storage_buffer storage_buffers[SG_MAX_STORAGEBUFFER_BINDSLOTS];
    sg_shader_image images[SG_MAX_IMAGE_BINDSLOTS];
    sg_shader_sampler samplers[SG_MAX_SAMPLER_BINDSLOTS];
    sg_shader_image_sampler_pair image_sampler_pairs[SG_MAX_IMAGE_SAMPLER_PAIRS];
    const char* label;
    uint32_t _end_canary;
} sg_shader_desc;
```

while the current V's is
https://github.com/vlang/v/blob/master/thirdparty/sokol/sokol_gfx.h#L3026C1
```c
typedef struct sg_shader_desc {
    uint32_t _start_canary;
    sg_shader_attr_desc attrs[SG_MAX_VERTEX_ATTRIBUTES];
    sg_shader_stage_desc vs;
    sg_shader_stage_desc fs;
    const char* label;
    uint32_t _end_canary;
} sg_shader_desc;
```

How often V language is expected to keep the pace with sokol updates?

I was using `import sokol.gfx` and worked but the API broke and then though rewriting some structs could be enough but soon I realize there is a lot of dependence including `gg`.

**Comment by JalonSolov** [comment, marked as answer]
There is no expectation of V keeping pace with sokol updates.  They happen when someone interested enough in doing so submits a PR.
